packages/cli-repl/src/cli-repl.ts

```diff
@@ -309,7 +314,11 @@ class CliRepl implements MongoshIOProvider {
     if (willExecuteCommandLineScripts) {
       this.mongoshRepl.setIsInteractive(willEnterInteractiveMode);
       this.bus.emit('mongosh:start-loading-cli-scripts', { usesShellOption: !!this.cliOptions.shell });
-      await this.loadCommandLineFilesAndEval(commandLineLoadFiles, evalScripts);
+      const exitCode = await this.loadCommandLineFilesAndEval(commandLineLoadFiles, evalScripts);
+      if (exitCode !== 0) {
+        await this.exit(exitCode);
```

Comment:
I'm not too familiar with this code, would exiting earlier here have any side effects that didn't happen before?

Reply:
So … in practice, `this.exit()` translates into doing some cleanup (e.g. flushing logs and telemetry), then `process.exit()`, so it should not really have any side effects beyond that. But maybe I’m not quite understanding the question here?

Reply:
All good, gave it a bit more of a read, I misunderstood.